src/binary/timestamp.rs

```diff
@@ -42,10 +42,18 @@ where
         // to derive the localized time.
         let utc = timestamp.date_time;
 
-        // Ion encodes offsets in minutes while DateTime stores it in seconds.
-        let offset_seconds = timestamp.offset.map(|o| o.utc_minus_local()).unwrap_or(0);
-        let offset_minutes = (offset_seconds as f32 / SECONDS_PER_MINUTE).round() as i64;
-        VarInt::write_i64(self, offset_minutes)?;
+        // Write out the offset (minutes difference from UTC). If the offset is
+        // unknown, negative zero is used.
+        if let Some(offset) = timestamp.offset {
+            // Ion encodes offsets in minutes while DateTime stores it in seconds.
+            let offset_seconds = offset.local_minus_utc();
```

Comment:
⚠️⚠️⚠️

I changed from `utc_minus_local()` to `local_minus_utc()`. I think this is right, but please check me.

Reply:
I think this is related to #190 it is encoded the "wrong way" in our `Timestamp` type.

Reply:
I see. Can I leave this as is in this PR and fix #190 as a follow up? I'm comfortable doing that since we have test coverage that ensures we don't get into an inconsistent state.

Reply:
Sure--I am just pointing out that your change is reasonable given #190, I might put a reference to that in the comment here as a TODO, but I am not too worried because, as you point out, a fix here will break a bunch of tests anyway.